src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js

```diff
@@ -58,7 +58,7 @@ export function* layoutManager({title, views='tables | images | xyPlots'}) {
                 break;
 
             case TABLE_LOADED :
-                [showImages, images] = handleNewTable(action, images, showImages);
+                [showImages, images] = handleNewTable(action, images, showImages, showTables);
```

Comment:
I don't see an action that say an XyPlot is added to the results.  I only see a TABLE_LOADED action.  So, use TABLE_LOADED to track when XyPlot.  You may consider adding actions to define specific chart's life-cycle.